cpp/inc/bond/core/config.h

```diff
@@ -40,3 +40,9 @@
 #define BOND_CONSTEXPR          BOOST_CONSTEXPR
 #define BOND_CONSTEXPR_OR_CONST BOOST_CONSTEXPR_OR_CONST
 #define BOND_STATIC_CONSTEXPR   BOOST_STATIC_CONSTEXPR
+
+#ifdef BOND_HEADER_ONLY
+#define BOND_HEADER_ONLY_INLINE inline
```

Comment:
`BOND_HEADER_ONLY` is intended to be a public-facing macro, but `BOND_HEADER_ONLY_INLINE` is a helper for us. Consider `BOND_DETAIL_HEADER_ONLY_INLINE`?

Reply:
Agree, but none of the rest is for public use either I suppose? :)

Reply:
Opened https://github.com/Microsoft/bond/issues/721 for this. I found one other public one, `BOND_DEFINE_BUFFER_MAGIC` and I could be convinced that `BOND_VERSION` and `BOND_MIN_CODEGEN_VERSION` are public.